### PR TITLE
logs/tailers/journald: fix initial seek

### PR DESCRIPTION
**Pull request by at-wat**
### What does this PR do?

When journald log `start_position` is set to `forceEnd`, all old logs are uploaded.
This commit fixes usage of sdjounal library to make `start_position` setting works.

### Motivation

- See #20416

### Additional Notes

none

### Possible Drawbacks / Trade-offs

none

### Describe how to test/QA your changes

1. Set journald log `start_position` config to `forceEnd`
2. Start datadog-agent
3. Check uploaded logs doesn't contain lines before the agent start time

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/logs/tailers/journald/tailer.go [hunk 1]
```diff
@@ -197,11 +197,26 @@ func (t *Tailer) forwardMessages() {
 func (t *Tailer) seek(cursor string) error {
 	mode, _ := config.TailingModeFromString(t.source.Config.TailingMode)
 
+	seekHead := func() error {
+		if err := t.journal.SeekHead(); err != nil {
+			return err
+		}
+		_, err := t.journal.Next() // SeekHead must be followed by Next
```
Comment (at-wat) [inline pkg/logs/tailers/journald/tailer.go:204]: https://pkg.go.dev/github.com/coreos/go-systemd/v22@v22.5.0/sdjournal#Journal.SeekHead
> SeekHead seeks to the beginning of the journal, i.e. the oldest available entry. This call **must be followed by a call to Next** before any call to Get* will return data about the first element.

File: pkg/logs/tailers/journald/tailer.go [hunk 2]
```diff
@@ -197,11 +197,26 @@ func (t *Tailer) forwardMessages() {
...
+		return err
+	}
+	seekTail := func() error {
+		if err := t.journal.SeekTail(); err != nil {
+			return err
+		}
+		_, err := t.journal.Previous() // SeekTail must be followed by Previous
```
Comment (at-wat) [inline pkg/logs/tailers/journald/tailer.go:211]: https://pkg.go.dev/github.com/coreos/go-systemd/v22@v22.5.0/sdjournal#Journal.SeekTail
> SeekTail may be used to seek to the end of the journal, i.e. the most recent available entry. This call **must be followed by a call to Previous** before any call to Get* will return data about the last element.

File: releasenotes/notes/fixes-journald-log-seek-on-init-d0d6465c8567916c.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fix journald log seek on initialization to beginning or end.
```
Comment (maycmlee) [inline releasenotes/notes/fixes-journald-log-seek-on-init-d0d6465c8567916c.yaml:11]:
```suggestion
    Fix journald log's Seek function to be set at the beginning or end upon initialization.
```

**Review by maycmlee — COMMENTED** [review]
Left some suggestions. Let me know if the meaning is correct.

**Review by gh123man — APPROVED** [review]
Thank you for the contribution! LGTM
Once the pending doc changes are addressed, ill merge this.

**Comment by at-wat** [comment]
@gh123man addressed the suggestion from @maycmlee

**Review by maycmlee — APPROVED** [review]

**Comment by gh123man** [comment]
Hi @at-wat, can you merge main into your branch please? There was a problem with our CI that is fixed now. I just need to get the CI passing before we merge this. 

Thanks

**Comment by at-wat** [comment]
@gh123man merged main into my branch

**Pull request merged**